### ISSUE TITLE
An error occurred after using the ”dp train --init-model model.ckpt init_model.json “command

### ISSUE THREAD

**Issue by XuFanffei** [deepmodeling/deepmd-kit#2799]
### Summary

An error ：
Errors may have originated from an input operation.
Input Source operations connected to node save/RestoreV2:
In[0] save/Const:	
In[1] save/RestoreV2/tensor_names:	
In[2] save/RestoreV2/shape_and_slices:
occurred after using the ”dp train --init-model model.ckpt init_model.json “command

init-model.json :
{   
    "model": {
        "type_map": [
            "N",
            "Al"
        ],
        "use_srtab": "AlN_Tab",
        "smin_alpha": 0.1,
        "sw_rmin":1.1,
        "sw_rmax":1.2,
        "descriptor": {
            "type": "se_e2_a",
            "sel": [
                90,
                90
            ],
            "rcut_smth": 0.5,
            "rcut": 6.0,
            "neuron": [
                25,
                50,
                100
            ],
            "resnet_dt": false,
            "axis_neuron": 12,
            "seed": 1729626010
        },
        "fitting_net": {
            "neuron": [
                240,
                240,
                240
            ],
            "resnet_dt": true,
            "seed": 1654538236
        }
    },
    "learning_rate": {
        "type": "exp",
        "start_lr": 0.00001,
        "decay_steps": 20000,
        "stop_lr": 1e-08
    },
    "loss": {
        "start_pref_e": 0.9,
        "limit_pref_e": 2,
        "start_pref_f": 100,
        "limit_pref_f": 1,
        "start_pref_v": 0.9,
        "limit_pref_v": 1
    },
    "training": {
        "stop_batch": 10,
        "disp_file": "lcurve_ZBL.out",
        "disp_freq": 2000,
        "numb_test": 2,
        "save_freq": 2000,
        "save_ckpt": "model.ckpt",
        "disp_training": true,
        "time_training": true,
        "profiling": false,
        "profiling_file": "timeline.json",
        "_comment": "that's all",
        "training_data": {
            "systems": [
                "data.init/auto_test/AlN/DFT/N_atom_magnetism/1176263/dpmd",
                "data.init/auto_test/AlN/DFT/Al_atom_magnetism/1176306/dpmd",
                "data.init/2h/2h.02x02x02/02.md/sys-0016-0016/deepmd",
                "data.init/3c/3c.02x03x03/02.md/sys-0018-0018/deepmd",
                "data.init/NaCl/NaCl.03x03x03/02.md/sys-0027-0027/deepmd",
                "data.init/P23/P23.02x03x02/02.md/sys-0024-0024/deepmd",
                "data.iters/iter.000000/02.fp/data.000",
                "data.iters/iter.000000/02.fp/data.008",
                "data.iters/iter.000000/02.fp/data.016",
                "data.iters/iter.000000/02.fp/data.024",
                "data.iters/iter.000001/02.fp/data.009",
                "data.iters/iter.000001/02.fp/data.017",
                "data.iters/iter.000001/02.fp/data.025",
                "data.iters/iter.000004/02.fp/data.004",
                "data.iters/iter.000004/02.fp/data.012",
                "data.iters/iter.000004/02.fp/data.028",
                "data.iters/iter.000005/02.fp/data.021",
                "data.iters/iter.000006/02.fp/data.014",
                "data.iters/iter.000006/02.fp/data.022",
                "data.iters/iter.000006/02.fp/data.030",
                "data.iters/iter.000007/02.fp/data.023",
                "data.iters/iter.000008/02.fp/data.192",
                "data.iters/iter.000008/02.fp/data.200",
                "data.iters/iter.000008/02.fp/data.208",
                "data.iters/iter.000008/02.fp/data.216",
                "data.iters/iter.000008/02.fp/data.224",
                "data.iters/iter.000008/02.fp/data.232",
                "data.iters/iter.000008/02.fp/data.240",
                "data.iters/iter.000008/02.fp/data.248",
                "data.iters/iter.000008/02.fp/data.256",
                "data.iters/iter.000008/02.fp/data.264",
                "data.iters/iter.000008/02.fp/data.272",
                "data.iters/iter.000008/02.fp/data.280",
                "data.iters/iter.000008/02.fp/data.288",
                "data.iters/iter.000008/02.fp/data.296",
                "data.iters/iter.000008/02.fp/data.304",
                "data.iters/iter.000008/02.fp/data.312",
                "data.iters/iter.000009/02.fp/data.193",
                "data.iters/iter.000009/02.fp/data.201",
                "data.iters/iter.000009/02.fp/data.209",
                "data.iters/iter.000009/02.fp/data.217",
                "data.iters/iter.000009/02.fp/data.225",
                "data.iters/iter.000009/02.fp/data.233",
                "data.iters/iter.000009/02.fp/data.241",
                "data.iters/iter.000009/02.fp/data.249",
                "data.iters/iter.000009/02.fp/data.257",
                "data.iters/iter.000009/02.fp/data.265",
                "data.iters/iter.000009/02.fp/data.273",
                "data.iters/iter.000009/02.fp/data.281",
                "data.iters/iter.000009/02.fp/data.289",
                "data.iters/iter.000009/02.fp/data.297",
                "data.iters/iter.000009/02.fp/data.305",
                "data.iters/iter.000009/02.fp/data.313",
                "data.iters/iter.000010/02.fp/data.194",
                "data.iters/iter.000010/02.fp/data.202",
                "data.iters/iter.000010/02.fp/data.210",
                "data.iters/iter.000010/02.fp/data.218",
                "data.iters/iter.000010/02.fp/data.226",
                "data.iters/iter.000010/02.fp/data.234",
                "data.iters/iter.000010/02.fp/data.242",
                "data.iters/iter.000010/02.fp/data.250",
                "data.iters/iter.000010/02.fp/data.258",
                "data.iters/iter.000010/02.fp/data.266",
                "data.iters/iter.000010/02.fp/data.274",
                "data.iters/iter.000010/02.fp/data.282",
                "data.iters/iter.000010/02.fp/data.290",
                "data.iters/iter.000010/02.fp/data.298",
                "data.iters/iter.000010/02.fp/data.306",
                "data.iters/iter.000010/02.fp/data.314",
                "data.iters/iter.000011/02.fp/data.195",
                "data.iters/iter.000011/02.fp/data.203",
                "data.iters/iter.000011/02.fp/data.211",
                "data.iters/iter.000011/02.fp/data.219",
                "data.iters/iter.000011/02.fp/data.227",
                "data.iters/iter.000011/02.fp/data.235",
                "data.iters/iter.000011/02.fp/data.243",
                "data.iters/iter.000011/02.fp/data.251",
                "data.iters/iter.000011/02.fp/data.259",
                "data.iters/iter.000011/02.fp/data.267",
                "data.iters/iter.000011/02.fp/data.275",
                "data.iters/iter.000011/02.fp/data.283",
                "data.iters/iter.000011/02.fp/data.291",
                "data.iters/iter.000011/02.fp/data.299",
                "data.iters/iter.000011/02.fp/data.307",
                "data.iters/iter.000011/02.fp/data.315",
                "data.iters/iter.000012/02.fp/data.196",
                "data.iters/iter.000012/02.fp/data.204",
                "data.iters/iter.000012/02.fp/data.212",
                "data.iters/iter.000012/02.fp/data.220",
                "data.iters/iter.000012/02.fp/data.228",
                "data.iters/iter.000012/02.fp/data.236",
                "data.iters/iter.000012/02.fp/data.244",
                "data.iters/iter.000012/02.fp/data.252",
                "data.iters/iter.000012/02.fp/data.260",
                "data.iters/iter.000012/02.fp/data.268",
                "data.iters/iter.000012/02.fp/data.276",
                "data.iters/iter.000012/02.fp/data.284",
                "data.iters/iter.000012/02.fp/data.292",
                "data.iters/iter.000012/02.fp/data.300",
                "data.iters/iter.000012/02.fp/data.308",
                "data.iters/iter.000012/02.fp/data.316",
                "data.iters/iter.000013/02.fp/data.197",
                "data.iters/iter.000013/02.fp/data.205",
                "data.iters/iter.000013/02.fp/data.213",
                "data.iters/iter.000013/02.fp/data.221",
                "data.iters/iter.000013/02.fp/data.229",
                "data.iters/iter.000013/02.fp/data.237",
                "data.iters/iter.000013/02.fp/data.245",
                "data.iters/iter.000013/02.fp/data.253",
                "data.iters/iter.000013/02.fp/data.261",
                "data.iters/iter.000013/02.fp/data.269",
                "data.iters/iter.000013/02.fp/data.277",
                "data.iters/iter.000013/02.fp/data.285",
                "data.iters/iter.000013/02.fp/data.293",
                "data.iters/iter.000013/02.fp/data.301",
                "data.iters/iter.000013/02.fp/data.309",
                "data.iters/iter.000013/02.fp/data.317",
                "data.iters/iter.000014/02.fp/data.198",
                "data.iters/iter.000014/02.fp/data.206",
                "data.iters/iter.000014/02.fp/data.214",
                "data.iters/iter.000014/02.fp/data.222",
                "data.iters/iter.000014/02.fp/data.230",
                "data.iters/iter.000014/02.fp/data.238",
                "data.iters/iter.000014/02.fp/data.246",
                "data.iters/iter.000014/02.fp/data.254",
                "data.iters/iter.000014/02.fp/data.262",
                "data.iters/iter.000014/02.fp/data.270",
                "data.iters/iter.000014/02.fp/data.278",
                "data.iters/iter.000014/02.fp/data.286",
                "data.iters/iter.000014/02.fp/data.294",
                "data.iters/iter.000014/02.fp/data.302",
                "data.iters/iter.000014/02.fp/data.310",
                "data.iters/iter.000014/02.fp/data.318",
                "data.iters/iter.000015/02.fp/data.199",
                "data.iters/iter.000015/02.fp/data.207",
                "data.iters/iter.000015/02.fp/data.215",
                "data.iters/iter.000015/02.fp/data.223",
                "data.iters/iter.000015/02.fp/data.231",
                "data.iters/iter.000015/02.fp/data.239",
                "data.iters/iter.000015/02.fp/data.247",
                "data.iters/iter.000015/02.fp/data.255",
                "data.iters/iter.000015/02.fp/data.263",
                "data.iters/iter.000015/02.fp/data.271",
                "data.iters/iter.000015/02.fp/data.279",
                "data.iters/iter.000015/02.fp/data.287",
                "data.iters/iter.000015/02.fp/data.295",
                "data.iters/iter.000015/02.fp/data.303",
                "data.iters/iter.000015/02.fp/data.311",
                "data.iters/iter.000015/02.fp/data.319",
                "data.iters/iter.000016/02.fp/data.320",
                "data.iters/iter.000016/02.fp/data.322",
                "data.iters/iter.000016/02.fp/data.324",
                "data.iters/iter.000016/02.fp/data.326",
                "data.iters/iter.000017/02.fp/data.321",
                "data.iters/iter.000017/02.fp/data.323",
                "data.iters/iter.000017/02.fp/data.325",
                "data.iters/iter.000017/02.fp/data.327",
                "data.iters/iter.000018/02.fp/data.328",
                "data.iters/iter.000018/02.fp/data.330",
                "data.iters/iter.000018/02.fp/data.332",
                "data.iters/iter.000018/02.fp/data.334",
                "data.iters/iter.000019/02.fp/data.329",
                "data.iters/iter.000019/02.fp/data.331",
                "data.iters/iter.000019/02.fp/data.333",
                "data.iters/iter.000019/02.fp/data.335",
                "data.iters/iter.000020/02.fp/data.336",
                "data.iters/iter.000020/02.fp/data.338",
                "data.iters/iter.000020/02.fp/data.340",
                "data.iters/iter.000020/02.fp/data.342",
                "data.iters/iter.000021/02.fp/data.337",
                "data.iters/iter.000021/02.fp/data.339",
                "data.iters/iter.000021/02.fp/data.341",
                "data.iters/iter.000021/02.fp/data.343",
                "data.iters/iter.000022/02.fp/data.344",
                "data.iters/iter.000022/02.fp/data.346",
                "data.iters/iter.000022/02.fp/data.348",
                "data.iters/iter.000022/02.fp/data.350",
                "data.iters/iter.000023/02.fp/data.345",
                "data.iters/iter.000023/02.fp/data.347",
                "data.iters/iter.000023/02.fp/data.349",
                "data.iters/iter.000023/02.fp/data.351",
                "data.iters/iter.000024/02.fp/data.352",
                "data.iters/iter.000024/02.fp/data.354",
                "data.iters/iter.000024/02.fp/data.356",
                "data.iters/iter.000024/02.fp/data.358",
                "data.iters/iter.000025/02.fp/data.353",
                "data.iters/iter.000025/02.fp/data.355",
                "data.iters/iter.000025/02.fp/data.357",
                "data.iters/iter.000025/02.fp/data.359"
            ],
            "batch_size": [
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                2,
                1,
                2,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                2,
                1,
                2,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                2,
                1,
                2,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                2,
                1,
                2,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                2,
                1,
                2,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                2,
                1,
                2,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                2,
                1,
                2,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                2,
                1,
                2,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                8,
                8,
                8,
                8,
                8,
                8,
                8,
                8,
                16,
                16,
                16,
                16,
                16,
                16,
                16,
                16,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1
            ]
        },
        "seed": 2868270054
    }
}

### DeePMD-kit Version

deepmd/2.1.0

### TensorFlow Version

null

### Details

[train.log](https://github.com/deepmodeling/deepmd-kit/files/12558263/train.log)

**Comment by njzjz** [comment]
Did you change the configuration compared to the checkpoint?

**Comment by taol1** [comment]
When I use the image: Registry. Dp. Tech/dptech/deepmd - kit: 2.2.3 - cuda11.6, appeared different error: the 2023-09-12 09:05:03. 420117: W tensorflow/core/framework/op_kernel.cc:1745] OP_REQUIRES failed at save_restore_v2_ops.cc:228 : NOT_FOUND:  Key fitting_attr/t_bias_atom_e not found in checkpoint

**Comment by njzjz** [comment]
When the model graph is changed, this error is expected. You can use the `-f ` argument to restore parameters from a frozen model, which does not require the same model structures.

**Comment by njzjz** [comment]
Feel free to reopen if the problem still exists. Thank you.

**Comment by chunweizhu** [comment]
When I used `dp train input.json --restart` I also encountered the same problem. Could it be due to a change in parallel parameters?

**Comment by njzjz** [comment]
When using `--restart` or `--init-model`, the model defination should not have any changes.
